### ISSUE TITLE
Configure rserver options via traitlets

### ISSUE THREAD

**Issue by rickmcgeer** [jupyterhub/jupyter-rsession-proxy#147]
### Proposed change

The documentation for [Jupyter Server Proxies](https://jupyter-server-proxy.readthedocs.io/en/latest/server-process.html) says that proxies can be configured by traitlets.  However, it appears that the rserver options are hardcoded in this proxy, specifically in [https://github.com/jupyterhub/jupyter-rsession-proxy/blob/master/jupyter_rsession_proxy/__init__.py](https://github.com/jupyterhub/jupyter-rsession-proxy/blob/master/jupyter_rsession_proxy/__init__.pl), lines 88-96:
```
cmd = [
            get_rstudio_executable('rserver'),
            '--auth-none=1',
            '--www-frame-origin=same',
            '--www-port=' + str(port),
            '--www-verify-user-agent=0',
            '--secure-cookie-key-file=' + ntf.name,
            '--server-user=' + get_system_user(),
        ]
```
We wanted to change `--www-frame-origin=same` to `---www-frame-origin=any`, appropriate for our application but not in general.  In order to do this, we forked the repo, made the change, and installed that.  Obviously, configuring this using traitlets would have been preferable.

### Alternative options

I don't know of any, but that may be ignorance (that is, there may be a better way to do this that I've simply missed, and any guidance would be appreciated).

### Who would use this feature?

Anyone who wants to launch rstudio from a JupyterHub, and tweak the environment.

### (Optional): Suggest a solution

Convert the body of this to a class, instantiate and configure the class using traitlets/a config file, and invoke the entry point command as a method on that class (or, alternatively, write a function which instantiates the object, applies the traitlets if any, then invokes the command and returns it).  I'll be happy to take this on, branch and do a PR if desired.  Any pointers to existing configurable proxies would make this easier on both me and the reviewers :-)

**Comment by ryanlovett** [comment]
Thanks @rickmcgeer ! I would love if you could submit a PR, perhaps to set the frame origin? For an example of using traitlet in such an extension, you can see an old, deprecated project of mine, https://github.com/ryanlovett/nbnovnc.

- [Traitlet](https://github.com/ryanlovett/nbnovnc/blob/master/nbnovnc/handlers.py#L34)
- [Instantiation](https://github.com/ryanlovett/nbnovnc/blob/master/nbnovnc/handlers.py#L49)
- [Reading the value](https://github.com/ryanlovett/nbnovnc/blob/master/nbnovnc/handlers.py#L71)

Then admins could set the value as `c.NBNoVNC.novnc_directory = "/usr/local/src/novnc"`.

**Comment by rickmcgeer** [comment]
Thanks, very much for the reply and encouragement @ryanlovett! I will get on it.  I may come back with questions.  Is this thread the right venue, or would you prefer something else?

**Comment by ryanlovett** [comment]
No problem if you have questions, feel free to ask here -- I may or may not be able answer :)  but I'll do my best.

If you make a PR and have questions about it, you can ask in that thread.

**Comment by rickmcgeer** [comment]
@ryanlovett I've been going over the code and running some experiments.  One thing I notice is that when I use this, and then enter the running container and kill the rserver process, it immediately restarts, which suggests that there's a daemon restarting it.  But that daemon doesn't seem to be in the usual spots (supervisor, etc) so I wondered where it was.  Also, in the Dockerfiles which use this, I don't see any RUN commands.  So it looks like this is all done in setup.py, in the structure:
`entry_points={
        'jupyter_serverproxy_servers': [
            'rstudio = jupyter_rsession_proxy:setup_rserver'
        ]
 },`
So my guess is that as I refactor this, I should leave setup.py as it is, and have `setup_rserver` return a string of the form that it currently does.  My approach right now is to move all of the subfunctions of `setup_rserver` into a subclass of `traitlets.config.SingletonConfigurable`, and `setup_rserver` will instantiate the sole class member and dig the command out of it.
Does this approach make sense?  Sorry for the noob question.

**Comment by ryanlovett** [comment]
jupyter-rsession-proxy uses jupyter-server-proxy which uses [simpervisor](https://github.com/jupyterhub/simpervisor) to start [supervised processes](https://github.com/jupyterhub/jupyter-server-proxy/blob/main/jupyter_server_proxy/handlers.py#L879). That is why the process gets restarted. See https://github.com/jupyterhub/jupyter-server-proxy for nearly everything that this project makes use of.

`setup_rserver` is the entry point for the proxied server so I don't think you'll be able to convert it to a traitlet object. It returns a dictionary ([server_process](https://github.com/jupyterhub/jupyter-rsession-proxy/blob/main/jupyter_rsession_proxy/__init__.py#L114)) and not a string. But you can turn the frame origin value into a traitlet that the command pulls in.

I hope this is helpful. I'm happy to follow up, and I appreciate the work you're putting into this!

**Comment by rickmcgeer** [comment]
@ryanlovett Thanks!  That's _exactly_ the information I was looking for!  Just FYI, I was planning to retain `setup_rserver` as the entry point, I just wanted to check to make sure I knew what it returned.
When I make some progress I'll be back to ask for feedback, and I really appreciate all the help you've given so far.

**Comment by rickmcgeer** [comment]
@ryanlovett I've made the change and issued a PR -- please take a look and tell me what you think.  I tried to make the changes minimal and of course tested before I did the push, but more eyes and tests are better than fewer!  I am quite new to this codebase, so it's entirely possible I missed something -- please let me know and I will fix it.

**Comment by ryanlovett** [comment]
Thanks @rickmcgeer ! I should be able to take a look at this later today.

**Comment by ryanlovett** [comment]
Fixed by #148.
